openbook_invitations/views.py

```diff
@@ -1,3 +1,139 @@
+
+        user = request.user
+        user_invites = []
+        if filter_status == INVITE_STATUS_ALL:
+            user_invites = user.get_user_invites().order_by('-pk')[offset:offset + count]
+        elif filter_status == INVITE_STATUS_PENDING:
+            user_invites = user.get_user_invites(status_pending=True).order_by('-pk')[offset:offset + count]
+        elif filter_status == INVITE_STATUS_ACCEPTED:
+            user_invites = user.get_user_invites(status_accepted=True).order_by('-pk')[offset:offset + count]
+
+        response_serializer = GetUserInviteSerializer(user_invites, many=True, context={"request": request})
+
+        return Response(response_serializer.data, status=status.HTTP_200_OK)
+
+
+class SearchUserInvites(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        query_params = request.query_params.dict()
+        serializer = SearchUserInvitesSerializer(data=query_params)
+        serializer.is_valid(raise_exception=True)
+
+        data = serializer.validated_data
+
+        count = data.get('count', 10)
+        query = data.get('query')
+        filter_status = data.get('status', INVITE_STATUS_ALL)
+
+        user = request.user
+        user_invites = []
+        if filter_status == INVITE_STATUS_ALL:
+            user_invites = user.search_user_invites().order_by('-pk')[:count]
+        elif filter_status == INVITE_STATUS_PENDING:
+            user_invites = user.search_user_invites(status_pending=True, query=query).order_by('-pk')[:count]
+        elif filter_status == INVITE_STATUS_ACCEPTED:
+            user_invites = user.search_user_invites(status_accepted=True, query=query).order_by('-pk')[:count]
+
+        response_serializer = GetUserInviteSerializer(user_invites, many=True, context={"request": request})
+
+        return Response(response_serializer.data, status=status.HTTP_200_OK)
+
+
+class UserInvite(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def delete(self, request, invite_id):
+        serializer = DeleteUserInviteSerializer(data={'invite_id': invite_id})
+        serializer.is_valid(raise_exception=True)
+
+        user = request.user
+
+        with transaction.atomic():
+            user.delete_user_invite_with_id(invite_id)
+
+        return ApiMessageResponse(_('Successfully deleted invite'), status=status.HTTP_200_OK)
+
+    def post(self, request, invite_id):
+        request_data = normalise_request_data(request.data)
+        request_data['invite_id'] = invite_id
+        serializer = EmailUserInviteSerializer(data=request_data)
+        serializer.is_valid(raise_exception=True)
+
+        invite_id = serializer.validated_data.get('invite_id')
+        email = serializer.validated_data.get('email')
+
+        user = request.user
+        with transaction.atomic():
+            try:
+                user.send_invite_to_invite_id_with_email(invite_id, email)
+            except Exception as e:
```

Comment:
This catch is too wide. Can catch database error, memory errors, everything. And we are disguising them all as `SMTPException`s.

Apart from that, does the SMTP exception get converted to an API error message with a status code as well? Otherwise this will just be a 500 error for the user.

Catch only the thing we are certain can go wrong when sending the invite and provide a nice error message for it.

If it's an `SMTPException`, catch only that, and raise a friendly Django rest exception for the requester.

Reply:
I thought that the `SMTPException` would get converted like the `ValidationErrors` do, but just checked and it doesnt. So will fix this. 

However, the problem with emails is multitude of things can go wrong, anything from the receiver SMTP server rejecting the email inbox, to their server being down to issues with AWS SES to something with Django, the whole email thing is a bit of a black box and I'm a bit lost what to catch and what not to. If we catch certain things, it might result in a 500 at other times.

Which is why to play it safe I did this, and wanted to log it but for now I print it since we dont have full logging in place. Ill look into this more on what can be done, pointers are welcome

Reply:
So for now, im catching the `SMTPException` and raising a `500` with a proper error message if anything goes wrong.